fw/t/unit/test_http_parser.c

```diff
@@ -103,6 +132,42 @@ split_and_parse_n(unsigned char *str, int type, size_t len, size_t chunks)
 	return r;
 }
 
+#else
+
+static int
+split_and_parse_n(unsigned char *str, int type, size_t len, size_t chunk_size)
+{
+	size_t pos = 0, step;
+	unsigned int parsed;
+	int r = 0;
+	TfwHttpMsg *hm = (type == FUZZ_REQ)
+			? (TfwHttpMsg *)req
+			: (TfwHttpMsg *)resp;
+
+	while (pos < len) {
```

Comment:
The problem of #1537 in this loop: instead of making only one iteration (as it was designed), we do this in the loop. The iteration variables must be moved to global context and the function must do only one iteration at a time.

We already have a bunch of global variables, so it'd be nice to organize them in a structure.

Also please remove PRIMES code and this will reduce of amount of required work and code, so you'll be able to fix #1537 in this PR.

Reply:
@krizhanovsky 

> The iteration variables must be moved to global context and the function must do only one iteration at a time.

The `split_and_parse_n()` function (both versions) seems to be quite correct in all many usage instancies except special cases of `fills_hdr_tbl_for_req()` and `fills_hdr_tbl_for_resp()`. I propose that these tests should not use the `split_and_parse_n()` at all. Instead, there should be a special "split_and_parse" function for these tests, which should involve neccessary test checks at every iteration over chunk sizes.

Reply:
@krizhanovsky 

> Also please remove PRIMES code and this will reduce of amount of required work and code, so you'll be able to fix #1537 in this PR.

No problem with it.
However, in this case we can not merge this PR before solving #1537.
Taking in account a high probability to find once more bug (probably, several ones) while solving #1537,
we probably never can merge this PR, never at all.

I would like yoi to acknoledge this decision once more time.